src/profiles_management/helpers/kfam.py

```diff
@@ -104,3 +217,85 @@ def list_contributor_authorization_policies(
         for ap in authorization_policies
         if has_kfam_annotations(ap) and not resource_is_for_profile_owner(ap)
     ]
+
+
+def kfam_resources_list_to_roles_dict(
+    resources: List[RoleBinding] | List[GenericNamespacedResource],
+) -> dict[str, List[classes.ContributorRole]]:
+    """Convert list of KFAM RoleBindings or AuthorizationPolicies to dict.
+
+    The user of the resource will be used as a key and its role as the value.
+
+    Args:
+        resources: List of KFAM RoleBindings or AuthorizationPolicies.
+
+    Returns:
+        Dictionary with keys the user names and values the roles, derived from parsing all
+        the provided resources.
+    """
+    contributor_roles_dict = {}
+    for resource in resources:
+        if has_kfam_annotations(resource):
+            user = get_contributor_user(resource)
+            role = get_contributor_role(resource)
+            contributor_roles_dict[user] = contributor_roles_dict.get(user, []) + [role]
+
+    return contributor_roles_dict
+
+
+def delete_rolebindings_not_matching_profile_contributors(
+    client: Client,
+    profile: classes.Profile,
+    existing_rolebindings: List[RoleBinding],
+) -> List[RoleBinding]:
+    """Delete RoleBindings in the cluster that doesn't match Contributors in PMR Profile.
+
+    Args:
+        client: The lightkube client to use.
+        profile: The PMR Profile to create RoleBindings based on its Contributors.
+        existing_rolebindings: RoleBindings in the cluster that will be evaluated for deletion.
+
+    Returns:
+        The remaining resources, after removing the deleted ones from the existing_resources.
+    """
+    role_bindings_to_delete = []
+    remaining_role_bindings = []
+
+    for rb in existing_rolebindings:
+        if not resource_matches_profile_contributor(rb, profile):
+            log.info(
+                "RoleBinding '%s' doesn't belong to Profile. Will delete it.",
+                k8s.get_name(rb),
+            )
+            role_bindings_to_delete.append(rb)
+        else:
+            remaining_role_bindings.append(rb)
+
+    log.info("Deleting all resources that don't match the PMR.")
+    delete_many(client, role_bindings_to_delete, logger=log)
+
+    return remaining_role_bindings
```

Comment:
I think that this method is more for operating rather than for returning. I think that after deleting the resources, one can use lightkube to actually see what resources are left using the K8s API, this is way more accurate than returning a list that we assume is correct.

Reply:
ACK! Did it mainly to avoid doing one more request for every Profile, but am fine doing it this way to improve readability and further simplify what the function does.

8495020